### PR TITLE
[wip] Implemented async version of initiateKeyTransfer

### PR DESCRIPTION
**Pull request by Jikstra**

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -386,10 +386,9 @@ test('ChatList methods', t => {
 })
 
 test('key transfer methods', t => {
-  t.comment('test234')
-  dc.initiateKeyTransfer(setupCode => {
+  dc.initiateKeyTransfer((err, setupCode) => {
+    t.ok(err == null, 'err is not null')
```
Comment (ralphtheninja) [inline test/index.js]: Better to do `t.same(err, null, 'err is not null')`

File: test/index.js [hunk 2]
```diff
@@ -385,6 +385,14 @@ test('ChatList methods', t => {
   t.end()
 })
 
+test('key transfer methods', t => {
+  dc.initiateKeyTransfer((err, setupCode) => {
+    t.ok(err == null, 'err is not null')
+    t.ok(setupCode != null, 'setupCode is null')
```
Comment (ralphtheninja) [inline test/index.js]: Also maybe do `t.is(typeof setupCode, 'string')`. Currently anything _but_ null is ok.

File: test/index.js (hunk 1 above)
Comment (ralphtheninja) [reply test/index.js]: The message as last parameter should say what the result should be, not what it shouldn't be.

**Comment by Jikstra** [comment]
Will close this as i rebased/merged this badly with master. New branch new luck.

**Pull request closed without merging**